cpp/include/raft/linalg/reduce_rows_by_key.cuh

```diff
@@ -108,6 +110,78 @@ void reduce_rows_by_key(const DataIteratorT d_A,
                      stream);
 }
 
+/**
+ * @defgroup reduce_rows_by_key Reduce Across Rows by Key
+ * @{
+ */
+
+/**
+ * @brief Computes the weighted reduction of matrix rows for each given key
```

Comment:
Same as above- can you put a todo in this file and reference a github issue to add an overload w/ generic reduction functions? I believe the weighted reduction here should be using a fused multiply-add intrinsic but a generalized reduction would make this primitive more widely applicable to different problems.